backend/src/main/java/com/google/sps/scraping/BBCGoodFoodRecipeScraper.java

```diff
@@ -102,38 +105,36 @@ private static String getCaloriesFromJson(JsonObject jsonObject) {
 
   /* Structure of jsonObject:
    * {suitableForDiet: "http://schema.org/VegetarianDiet, http://schema.org/GlutenFreeDiet .."} */
-  private static String[] getDietFromJson(JsonObject jsonObject) {
+  private static List<String> getDietFromJson(JsonObject jsonObject) {
     JsonElement dietElements = jsonObject.get("suitableForDiet");
     if (dietElements == null) {
-      return new String[0];
+      return new ArrayList<>();
     }
-    String[] diet = dietElements.getAsString().split(", ");
+    List<String> diet = Arrays.asList(dietElements.getAsString().split(", "));
     int counter = 0;
     for (String item : diet) {
-      diet[counter++] = item.replaceAll("http://schema.org/|Diet", "").toLowerCase();
+      diet.set(counter++, item.replaceAll("http://schema.org/|Diet", "").toLowerCase());
```

Comment:
I think this turned out a bit weird. The `counter` isn't really used for any reason other than putting the element back in the same position. I think this might look better:

```
List<String> diet = Arrays.asList(dietElements.getAsString().split(", "));
for (int i = 0; i < diet.size(); i++) {
  diet.set(i, diet.get(i).replaceAll("http://schema.org/|Diet", "").toLowerCase());
}
```

Or even this:
```
List<String> diet = new List<>();
for (String item : dietElements.getAsString().split(", ")) {
  diet.add(item.replaceAll("http://schema.org/|Diet", "").toLowerCase());
}
```

WDYT?